utils/auth.ts

```diff
@@ -0,0 +1,31 @@
+import { AUTH_COOKIE_KEYS } from '@/apis/config/type';
+import { CookieKeyType } from '@/types';
+
+const generateCookiesKeyValues = ({ accessToken, refreshToken, userId }: CookieKeyType) => {
+  const accessTokenExpireDate = new Date();
+
+  return [
+    [AUTH_COOKIE_KEYS.accessToken, accessToken],
+    [AUTH_COOKIE_KEYS.refreshToken, refreshToken],
+    [AUTH_COOKIE_KEYS.userId, userId],
+    [AUTH_COOKIE_KEYS.accessTokenExpireDate, accessTokenExpireDate.getTime()],
+  ];
+};
+
+const getAuthTokensByCookie = (cookieString: string): Partial<CookieKeyType> => {
+  const auth: Partial<CookieKeyType> = {};
+  for (const cookie of cookieString.split('; ')) {
+    const [key, value] = cookie.split('=');
+    if (key === AUTH_COOKIE_KEYS.accessToken) {
+      auth.accessToken = value;
+    } else if (key === AUTH_COOKIE_KEYS.refreshToken) {
+      auth.refreshToken = value;
+    } else if (key === AUTH_COOKIE_KEYS.userId) {
+      auth.userId = +value;
+    }
+  }
+  return auth;
+};
```

Comment:
for of 문은 사이드 이펙트가 생길 수 있어 보다 안전한 고차함수를 활용하는게 좋아보여요!

이런 방법은 어떨까요? 더 나은 방법이 있으면 바꿔도 좋을 것 같아용

```suggestion
cookieString.split('; ').forEach((cookie) => {
  const [key, value] = cookie.split('=');
  if (Object.values(AUTH_COOKIE_KEYS).includes(key)) {
    if (key === AUTH_COOKIE_KEYS.userId) {
      cookies[key] = +value;
    } else {
      cookies[key] = value;
    }
  }
});
```

Reply:
`for of`문은 Javascript에서 배열을 순회하는 가장 강력한 방법으로, `for/in`, `for`, `forEach`보다 더 많은 장점을 가지고 있습니다. 

1. [await를 사용한 `for-await-of`](https://exploringjs.com/impatient-js/ch_async-iteration.html#for-await-of)
2. `break`, `continue`사용

for/of문은 forEach과 달리 반복을 중단을 할 수 있다는 점이 있지만, 이러한 점이 이 상황에서 사이드 이펙트를 일으킬 수 없다고 생각합니다.

어떤 점에서 사이드 이펙트가 일어날 수 있다고 생각하셨는 지 궁금합니다.

- [참고 : thecodebabarian blog](https://thecodebarbarian.com/for-vs-for-each-vs-for-in-vs-for-of-in-javascript.html)